flower_classifier/train.py

```diff
@@ -14,11 +15,16 @@
 @hydra.main(config_path="../conf", config_name="config")
 def train(cfg):
     data_module = hydra.utils.instantiate(cfg.dataset)
+
+    lr_scheduler = getattr(cfg, "lr_scheduler", None)
+    scheduler = getattr(lr_scheduler, "scheduler", None)
+    steps_per_epoch = getattr(scheduler, "steps_per_epoch", None)
+    if steps_per_epoch == "AUTO":
+        data_module.prepare_data()
+        lr_scheduler.scheduler.steps_per_epoch = math.ceil(data_module.len_train / cfg.dataset.batch_size)
```

Comment:
let's wrap this into a function which accepts the `DictConfig` object and returned a modified version, that way we can write a couple tests to ensure this works as we intend.